libs/fundamental-redirects/index.js

```diff
@@ -1162,7 +1162,12 @@ const MISC_REDIRECT_PATTERNS = [
     // which is what we use today.
     colonToSlash: true,
   }),
-  localeRedirect(/^DOM[\/$]?/i, "/docs/DOM", { permanent: true }),
+  localeRedirect(
+    /^DOM(?<subPath>$|\/.*)/i,
+    ({ subPath }) =>
+      `/docs/DOM${subPath.endsWith("/") ? subPath.slice(0, -1) : subPath}`,
+    { permanent: true }
+  ),
```

Comment:
```suggestion
  localeRedirect(
    /^DOM(?<subPath>\/.+)?\/?$/i,
    ({ subPath = "" }) =>
      `/docs/DOM${subPath}`,
    { permanent: true }
  ),
```
should do the same

Reply:
What you have won't remove the trailing slash for cases like `/en-US/DOM/element.addEventListener/` since the `.+` is greedy, but when I make it non-greedy with `.+?` it works great and looks cleaner, thanks @fiji-flo !